src/utils/async-queue.ts

```diff
@@ -0,0 +1,42 @@
+export type QueueEntry = () => void;
+
+// queue manager is an class that helps with managing tasks.
+// More specifically, it helps with tasks that are often used.
+// It's fully asyncronous and uses promises and tries to get 60FPS.
+export default class AsyncQueue {
+    private queue: QueueEntry[] = [];
+    private timerID: number = null;
```

Comment:
There's a naming convention :) ... timer**Id**

Reply:
Really need to look into the naming-convetions rule from eslint 😄